### PR TITLE
Use requirements.yaml to define dependencies

### LINKED ISSUE #132: [Stable/Drupal] Update MariaDB dependency
This chart should use the new requirements.yaml file and update the MariaDB dependency to be the MariaDB chart in stable/MariaDB.

### PR DESCRIPTION
**Pull request by prydonius**
Updates incubator/patroni, stable/redmine, stable/wordpress,
and stable/drupal to vendor dependencies using Helm's built-in
dependency management.

This also updates the etcd dependency for incubator/patroni to 0.1.1, and mariadb for stable/redmine, stable/wordpress and stable/drupal to 0.5.2.

closes #132 #131

### CODE REVIEW

**Comment by viglesiasce** [comment]
@k8s-bot test this

**Comment by viglesiasce** [comment]
@k8s-bot test this

**Comment by prydonius** [comment]
Is GitHub having issues with hooks?

**Comment by prydonius** [comment]
@viglesiasce PTAL, tests are passing now!

**Comment by viglesiasce** [comment]
@prydonius do we need to implement anything in the repo-sync script in order for this to work in the public chart repos?

**Comment by viglesiasce** [comment]
Ah i see you checked in the tarball. I would prefer that we not do that as to not make the repository unwieldy. I would prefer to do the `dependency update` during repo sync time.

**Comment by prydonius** [comment]
@viglesiasce makes sense, will remove the tarballs. Should we update the `.gitignore` to ignore `charts/*.tgz`?

**Comment by viglesiasce** [comment]
Yep ignoring tarballs inside the chart makes sense.

**Comment by prydonius** [comment]
@viglesiasce okay removed the vendored deps, updated `.gitignore` and updated test and repo sync scripts to pull dependencies

File: test/changed.sh [hunk 1]
```diff
@@ -45,13 +45,15 @@ gcloud container clusters get-credentials jenkins --project kubernetes-charts-ci
...
 HELM_TARBALL=helm-canary-linux-amd64.tar.gz
+INCUBATOR_REPO_URL=http://storage.googleapis.com/kubernetes-charts-incubator
 pushd /opt
   wget -q ${HELM_URL}/${HELM_TARBALL}
   tar xzfv ${HELM_TARBALL}
   PATH=/opt/linux-amd64/:$PATH
 popd
 
 helm init --client-only
+helm repo add incubator ${INCUBATOR_REPO_URL}
```
Comment (prydonius) [inline test/changed.sh:56]: `helm dep update` complains if the repository URL isn't added. This has the nice side-effect of allowing us to enforce which repositories a chart can pull from.

**Comment by viglesiasce** [comment]
Awesome. Testing the patroni chart now.

**Comment by viglesiasce** [comment]
Patroni chart is failing as follows:

``` console
2016-10-21 23:48:26,717 INFO: waiting on etcd
2016-10-21 23:48:31,816 ERROR: Can not resolve SRV for willing-lionfi-etcd.%!s(<nil>).svc.cluster.local
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/patroni/dcs/etcd.py", line 150, in get_srv_record
    return [(str(r.target).rstrip('.'), r.port) for r in resolver.query('_etcd-server._tcp.' + host, 'SRV')]
  File "/usr/local/lib/python3.4/dist-packages/dns/resolver.py", line 1068, in query
    raise_on_no_answer, source_port)
  File "/usr/local/lib/python3.4/dist-packages/dns/resolver.py", line 995, in query
    raise NXDOMAIN(qname=qnames_to_try)
dns.resolver.NXDOMAIN: None of DNS query names exist: ['_etcd-server._tcp.willing-lionfi-etcd.%!s\\(<nil>\\).svc.cluster.local.', '_etcd-server._tcp.willing-lionfi-etcd.%!s\\(<nil>\\).svc.cluster.local.default.svc.cluster.local.', '_etcd-server._tcp.willing-lionfi-etcd.%!s\\(<nil>\\).svc.cluster.local.svc.cluster.local.', '_etcd-server._tcp.willing-lionfi-etcd.%!s\\(<nil>\\).svc.cluster.local.cluster.local.', '_etcd-server._tcp.willing-lionfi-etcd.%!s\\(<nil>\\).svc.cluster.local.google.internal.', '_etcd-server._tcp.willing-lionfi-etcd.%!s\\(<nil>\\).svc.cluster.local.c.vic-goog.internal.']
2016-10-21 23:48:31,822 INFO: waiting on etcd
```

**Comment by viglesiasce** [comment]
This line in the petset definition is still trying to use the Namespace value:
https://github.com/kubernetes/charts/blob/master/incubator/patroni/templates/ps-patroni.yaml#L47

**Comment by prydonius** [comment]
my bad! Good catch, does the change I just pushed fix it? My cluster doesn't support PetSets.

**Comment by viglesiasce** [comment]
Yep patroni looks good. Moving on to other charts.

**Comment by viglesiasce** [comment]
Tested redmine, drupal and wordpress and they all seem good to go.

**Pull request merged**